spec/requests/custom_buttons_spec.rb

```diff
@@ -0,0 +1,206 @@
+RSpec.describe 'CustomButtons API' do
+  let(:object_def) { FactoryGirl.create(:generic_object_definition, :name => 'foo') }
+  let(:cb) { FactoryGirl.create(:custom_button, :name => 'custom_button', :applies_to_class => 'GenericObjectDefinition', :applies_to_id => object_def.id) }
+
+  describe 'GET /api/custom_buttons' do
+    it 'does not list custom buttons without an appropriate role' do
+      api_basic_authorize
+
+      get(api_custom_buttons_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'lists all custom buttons with an appropriate role' do
+      api_basic_authorize collection_action_identifier(:custom_buttons, :read, :get)
+      cb_href = api_custom_button_url(nil, cb)
+
+      get(api_custom_buttons_url)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'custom_buttons',
+        'resources' => [
+          hash_including('href' => cb_href)
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'GET /api/custom_buttons/:id' do
+    it 'does not let you query custom buttons without an appropriate role' do
+      api_basic_authorize
+
+      get(api_custom_button_url(nil, cb))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'can query a custom button by its id' do
+      api_basic_authorize action_identifier(:custom_buttons, :read, :resource_actions, :get)
+
+      get(api_custom_button_url(nil, cb))
+
+      expected = {
+        'id'   => cb.id.to_s,
+        'name' => "custom_button"
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'POST /api/custom_buttons' do
+    it 'can create a new custom button' do
+      api_basic_authorize collection_action_identifier(:custom_buttons, :create)
+
+      cb_rec = {
+        'name'             => 'Generic Object Custom Button',
+        'description'      => 'Generic Object Custom Button description',
+        'applies_to_class' => 'GenericObjectDefinition',
+        'options'          => {
```

Comment:
Can you add a resource action to this request just to prove that that works?

Reply:
in 5104806